pkg/appliance/backup.go

```diff
@@ -246,6 +246,16 @@ func PerformBackup(cmd *cobra.Command, args []string, opts *BackupOpts) (map[str
 		if err := retryStatus(ctx, b.applianceID, b.backupID); err != nil {
 			return b, err
 		}
+		if opts.Config.Version >= 17 {
+			// Check for backup result and output implemented in API version 17
+			result, output, err := backupAPI.Result(ctx, b.applianceID, b.backupID)
+			if err != nil {
+				return b, err
+			}
+			if result != backup.Success {
+				return b, errors.New(output)
```

Comment:
I'd rather see that refactor backup.Status instead of .Result() method, since **output** can be empty, this error message does not really make since.

how about
```golang
	retryStatus := func(ctx context.Context, applianceID, backupID string) error {
		bo := backoff.NewExponentialBackOff()
		bo.MaxElapsedTime = 0
		return backoff.Retry(func() error {
			status, err := backupAPI.Status(ctx, applianceID, backupID)
			if err != nil {
				return err
			}
			log.WithFields(log.Fields{
				"appliance":     applianceID,
				"backup_id":     backupID,
				"backup_status": status.GetStatus(),
			}).Infof("backup status %s", status.GetOutput()) // output as string, not field, since it can be empty

			if status.GetStatus() != backup.Done {
				return fmt.Errorf("Backup not done for appliance %s, got %s %s %s", applianceID, status.GetStatus(), status.GetOutput(), status.GetResult())
			}
			return nil
		}, bo)
	}
```


and backup.status
```diff
diff --git a/pkg/appliance/backup/backup.go b/pkg/appliance/backup/backup.go
index 5d3457a..87b8133 100644
--- a/pkg/appliance/backup/backup.go
+++ b/pkg/appliance/backup/backup.go
@@ -59,12 +59,12 @@ const (
 	Failure    string = "failure"
 )
 

-func (b *Backup) Status(ctx context.Context, applianceID, backupID string) (string, error) {
+func (b *Backup) Status(ctx context.Context, applianceID, backupID string) (*openapi.AppliancesIdBackupBackupIdStatusGet200Response, error) {
 	status, response, err := b.APIClient.ApplianceBackupApi.AppliancesIdBackupBackupIdStatusGet(ctx, applianceID, backupID).Authorization(b.Token).Execute()
 	if err != nil {
-		return "", api.HTTPErrorResponse(response, err)
+		return nil, api.HTTPErrorResponse(response, err)
 	}
-	return status.GetStatus(), nil
+	return status, nil
 }
 
 func (b *Backup) Result(ctx context.Context, applianceID, backupID string) (string, string, error) {
```

Reply:
I can refactor the `backup.Status`, but not this way because it would not work. We need to first check if the status is done _and then_ what the result is. Only checking the status is not enough since it can be done, but still have failed. That's what the result field is for.